t/20_sdaf_deployment_library.t

```diff
@@ -640,4 +659,30 @@ subtest '[sdaf_execute_playbook] Command verbosity' => sub {
     undef_variables();
 };
 
+subtest '[git_clone] Compose command' => sub {
+    my $ms_sdaf = Test::MockModule->new('sles4sap::sdaf_deployment_library', no_auto => 1);
+    my @calls;
+    $ms_sdaf->redefine(assert_script_run => sub { @calls = $_[0]; return 0; });
+    $ms_sdaf->redefine(record_info => sub { return 0; });
+    my $repository = 'https://github.com/os-autoinst/os-autoinst-distri-opensuse';
+
+    git_clone($repository);
+    is join(' ', @calls), "git clone $repository", "Check base command composition";
+
+    git_clone($repository, skip_ssl_verification => 'true', output_log_file => 'output.log', branch => 'world_domination');
+    note("\n  -->  " . join("\n  -->  ", @calls));
+    ok(grep(/git -c http.sslVerify=false clone/, @calls), 'Option "-c http.sslVerify=false " must be between "git" and "clone"');
```

Comment:
is there a problem here? you call the function with `skip_ssl_verification => 'true'` but then you grep with `false`. Does `grep` works with `@calls` that is a list? usually we use a different pattern like:

```
ok((any { /git.*clone.*github.*com\/SUSE\/qe-sap-deployment.*\/BRUCE/ } @calls), 'Default repo cloned in /BRUCE');
```

Reply:
Yeah.. there is a difference between the function parameter and git parameter logic. 
Git parameter is `http.sslVerify=false` to turn off the verification which is enabled by default.  
Perl function variable name is 'skip_ssl_verification' and you skip it with 'true' value. 

I could do it the same way but then you get into the weird area where caller does 
'git_clone(http_ssl_verify=>"false")' but perl interprets this as non empty string with 'true' value.

Reply:
Is value put `skip_ssl_verification` opposite to the one that is composed in the git command line?
If so can we avoid it? maybe using  argument named `ssl_verification` and at the beginning of the method add
```
$args{ssl_verification} =// 1;
```

and later 
```
$git_cmd = sed /git/git -http.sslVerify=false/ if not $args{ssl_verification};
```


So internally yes code is a little bit more complicate, but API is in direct logic: from the external user put nothing and get nothing on the git command, put 'false' and get 'false' in the git command.

Reply:
It is opposite but you don't need to map it. API caller does not need to care about internal working of the code. 
I really prefer to keep it that way and keep the code straightforward to understand even for code maintenance. 

With `$args{ssl_verification} =// 1;` if user defines parameter as `ssl_verification=>"false"` or `"0"` verification won't turn off. So you need to sanitize inputs as well. 

With `skip_ssl_verification` you only define parameter to skip it. 
I adjusted the description so it is even more clear:
```B<skip_ssl_verification>: Disable SSL verification. Can be useful in case of self signed certificates.
    Define this parameter B<ONLY> if you want to skip verification.```